sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_asset_utils.py

```diff
@@ -185,3 +194,34 @@ def test_directory_size_with_ignore_file(self, storage_test_directory: str, amli
         _, _ = generate_link_file(storage_test_directory)
         with_symlink_size = get_directory_size(storage_test_directory)
         assert len(with_symlink_size[1]) == len(base_size[1]) + 2
+
+    def test_check_or_modify_auto_delete_setting(self):
```

Comment:
add some e2e test for this feature?

Reply:
add a e2e test on sdk/ml/azure-ai-ml/tests/dataset/e2etests/test_data.py